packages/playwright-core/src/client/browser.ts

```diff
@@ -60,12 +61,14 @@ export class Browser extends ChannelOwner<channels.BrowserChannel> implements ap
 
   async newContext(options: BrowserContextOptions = {}): Promise<BrowserContext> {
     options = { ...this._browserType._defaultContextOptions, ...options };
+    const harRouter = options.har ? await HarRouter.create(options.har) : null;
     const contextOptions = await prepareBrowserContextParams(options);
     const context = BrowserContext.from((await this._channel.newContext(contextOptions)).context);
     context._options = contextOptions;
     this._contexts.add(context);
     context._logger = options.logger || this._logger;
     context._setBrowserType(this._browserType);
+    harRouter?.addRoute(context);
```

Comment:
There are more places that create context, e.g. `launchPersistentContext` or `electron`.